UTRAnnotator.pm

```diff
@@ -0,0 +1,1349 @@
+
+=cut
+
+
+package UTRAnnotator;
+
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepPlugin);
+use Bio::EnsEMBL::Utils::Sequence qw(reverse_comp);
+
+use Scalar::Util qw(looks_like_number);
+
+use strict;
+
+sub feature_types {
+  return ['Transcript'];
+}
+
+sub new {
+  my $class = shift;
+  my $self = $class->SUPER::new(@_);
+
+  printf "\nWarning: no FASTA file specified (--fasta). VEP running will take a long time." if ($self->config->{fasta} eq "");
+
+  my $param_hash = $self->params_to_hash();
+
+  if (-e $param_hash->{file}){
+    open my $fh, "<", $param_hash->{file} or die $!;
+    my %uORF_evidence;
+
+    while (<$fh>) {
+      chomp;
+      my ($chr, $pos, $gene, $strand, $type, $stop_pos) = split /\t/;
+      next if $chr eq "chr" && $pos eq "pos";
+
+      die "$chr does not exist; Check your reference file." unless ($chr =~ m/^(chr*)/);
+      die "Position $pos is not a number; Check your reference file." unless looks_like_number($pos);
+
+      my $key = $chr . ":" . $pos; # chr has 'chr' prefix
+      $uORF_evidence{$key} = 1;
+  }
+
+  close $fh;
+
+  $self->{uORF_evidence} = \%uORF_evidence;
+
+  } else {
+      printf "Warning: small ORF file not found. For human, you could use the curated list of uORFs found in the repository (https://github.com/Ensembl/UTRannotator):\n" .
+      "'uORF_starts_ends_GRCh37_PUBLIC.txt' for GRCh37 or 'uORF_starts_ends_GRCh38_PUBLIC.txt' for GRCh38.\n";
+  }
+
+  # Kozak-Strength Class
+  my %kozak_strength;
+  $kozak_strength{1}='Weak';
+  $kozak_strength{2}='Moderate';
+  $kozak_strength{3}='Strong';
+
+  $self->{kozak_strength} = \%kozak_strength;
+
+  return $self;
+}
+
+sub get_header_info {
+
+  my $self->{_header_info} = {
+        five_prime_UTR_variant_consequence => "Output the variant consequences of a given 5 prime UTR variant: 5_prime_UTR_premature_start_codon_gain_variant, " . 
+        "5_prime_UTR_stop_codon_gain_variant, 5_prime_UTR_premature_start_codon_loss_variant, 5_prime_UTR_stop_codon_loss_variant or 5_prime_UTR_uORF_frameshift_variant",
+        five_prime_UTR_variant_annotation => "Output the annotation of a given 5 prime UTR variant",
+        existing_uORFs => 'The number of existing uORFs with a stop codon within the 5 prime UTR',
+        existing_OutOfFrame_oORFs => 'The number of existing out-of-frame overlapping ORFs (OutOfFrame oORF) at the 5 prime UTR',
+        existing_InFrame_oORFs => 'The number of existing inFrame overlapping ORFs (inFrame oORF) at the 5 prime UTR',
+  };
+  return $self->{_header_info};
+}
+
+sub run {
+  my ($self, $tva) = @_;
+
+  #only annotate the effect if the variant is 5_prime_UTR_variant
+  return {} unless grep {$_->SO_term eq '5_prime_UTR_variant'}  @{$tva->get_all_OverlapConsequences};
+
+  my $bvfo = $tva->base_variation_feature_overlap;
```

Comment:
Suggesting that this is also removed as $tva can be used to access allele string and the transcript